chcemvediet/apps/inforequests/tests/test_models/test_action.py

```diff
@@ -8,265 +8,267 @@
+                (Action.TYPES.DISCLOSURE,             Deadline(Deadline.TYPES.APPLICANT_DEADLINE, delivered_date, 15, Deadline.UNITS.CALENDAR_DAYS, None), dict(disclosure_level=Action.DISCLOSURE_LEVELS.PARTIAL)),
+                (Action.TYPES.DISCLOSURE,             None, dict(disclosure_level=Action.DISCLOSURE_LEVELS.FULL)),
+                (Action.TYPES.REFUSAL,                Deadline(Deadline.TYPES.APPLICANT_DEADLINE, delivered_date, 15, Deadline.UNITS.CALENDAR_DAYS, None), dict()),
+                (Action.TYPES.AFFIRMATION,            None, dict()),
+                (Action.TYPES.REVERSION,              None, dict()),
+                (Action.TYPES.REMANDMENT,             Deadline(Deadline.TYPES.OBLIGEE_DEADLINE, workdays.advance(legal_date, 4), 8, Deadline.UNITS.WORKDAYS, None), dict()),
+                (Action.TYPES.ADVANCED_REQUEST,       Deadline(Deadline.TYPES.OBLIGEE_DEADLINE, workdays.advance(legal_date, 4), 8, Deadline.UNITS.WORKDAYS, None), dict()),
+                (Action.TYPES.EXPIRATION,             Deadline(Deadline.TYPES.APPLICANT_DEADLINE, legal_date, 15, Deadline.UNITS.CALENDAR_DAYS, None), dict()),
+                (Action.TYPES.APPEAL_EXPIRATION,      None, dict()),
                 )
         # Make sure we are testing all defined action types
         tested_action_types = set(a for a, _, _ in tests)
         defined_action_types = Action.TYPES._inverse.keys()
         self.assertItemsEqual(tested_action_types, defined_action_types)
 
-        inforequest = self._create_inforequest()
-        branch = self._create_branch(inforequest=inforequest)
         for action_type, expected_deadline, extra_args in tests:
-            action = self._create_action(branch=branch, type=action_type, omit=[u'deadline'], **extra_args)
-            self.assertEqual(action.deadline, expected_deadline)
+            previous = self._create_action(type=Action.TYPES.REQUEST, delivered_date=delivered_date)  # Deadline for CONFIRMATION, EXTENSION
+            action = self._create_action(type=action_type, delivered_date=delivered_date, legal_date=legal_date, **extra_args)
+            self.assertEqual(repr(action.deadline), repr(expected_deadline))
 
-    def test_deadline_field_is_not_reset_to_default_value_if_saving_existing_instance(self):
-        inforequest = self._create_inforequest()
-        branch = self._create_branch(inforequest=inforequest)
-        action = self._create_action(branch=branch, deadline=3)
+    def test_deadline_field_is_not_changed_if_saving_existing_instance(self):
+        action = self._create_action()
+        deadline = action.deadline
         action.subject = u'Changed'
         action.save()
         action = Action.objects.get(pk=action.pk)
-        self.assertEqual(action.deadline, 3)
+        self.assertEqual(repr(action.deadline), repr(deadline))
 
     def test_extension_field(self):
```

Comment:
Testy fieldov/metod/properies daj v rovnakom poradi ako su definovane v modeli. Teda deadline az niekde na konci. Povodne to bol field, preto deadline bol povodne tu hore.

Reply:
:ok: